src/shared/lib/comparison/ComparisonStore.ts

```diff
@@ -1181,71 +1230,44 @@ export default class ComparisonStore {
         {}
     );
 
-    readonly overallSurvivalDescriptions = remoteData({
+    readonly survivalDescriptions = remoteData({
         await: () => [
             this.activeStudiesClinicalAttributes,
             this.activeStudyIdToStudy,
+            this.survivalClinicalAttributesPrefix,
         ],
         invoke: () => {
-            const overallSurvivalClinicalAttributeId = 'OS_STATUS';
-            const clinicalAttributeMap = _.groupBy(
-                this.activeStudiesClinicalAttributes.result,
-                'clinicalAttributeId'
-            );
-            const result: ISurvivalDescription[] = [];
-            const studyIdToStudy: { [studyId: string]: CancerStudy } = this
-                .activeStudyIdToStudy.result;
-            if (
-                clinicalAttributeMap &&
-                clinicalAttributeMap[overallSurvivalClinicalAttributeId] &&
-                clinicalAttributeMap[overallSurvivalClinicalAttributeId]
-                    .length > 0
-            ) {
-                clinicalAttributeMap[
-                    overallSurvivalClinicalAttributeId
-                ].forEach(attr => {
-                    result.push({
-                        studyName: studyIdToStudy[attr.studyId].name,
-                        description: attr.description,
-                    } as ISurvivalDescription);
-                });
-                return Promise.resolve(result);
-            }
-            return Promise.resolve([]);
-        },
-    });
-
-    readonly diseaseFreeSurvivalDescriptions = remoteData({
-        await: () => [
-            this.activeStudiesClinicalAttributes,
-            this.activeStudyIdToStudy,
-        ],
-        invoke: () => {
-            const diseaseFreeSurvivalClinicalAttributeId = 'DFS_STATUS';
-            const clinicalAttributeMap = _.groupBy(
-                this.activeStudiesClinicalAttributes.result,
-                'clinicalAttributeId'
-            );
-            const result: ISurvivalDescription[] = [];
-            const studyIdToStudy: { [studyId: string]: CancerStudy } = this
-                .activeStudyIdToStudy.result;
-            if (
-                clinicalAttributeMap &&
-                clinicalAttributeMap[diseaseFreeSurvivalClinicalAttributeId] &&
-                clinicalAttributeMap[diseaseFreeSurvivalClinicalAttributeId]
-                    .length > 0
-            ) {
-                clinicalAttributeMap[
-                    diseaseFreeSurvivalClinicalAttributeId
-                ].forEach(attr => {
-                    result.push({
-                        studyName: studyIdToStudy[attr.studyId].name,
-                        description: attr.description,
-                    } as ISurvivalDescription);
+            const result: { [prefix: string]: ISurvivalDescription[] } = {};
+
+            if (this.survivalClinicalAttributesPrefix) {
```

Comment:
you don't have to test for existence of this.  it's the result which could be undefined.